Added support for break armour on hit

MrHB212 opened the pull request:
Currently it seems to work but i have to do some more testing. 

### After screenshot:
![image](https://github.com/user-attachments/assets/83f02824-e071-4c34-8656-edcc120501a5)

MrHB212 commented on src/Modules/CalcSections.lua:
```diff
@@ -624,6 +624,7 @@ return {
 	{ label = "ES % per second", color = colorCodes.ES, haveOutput = "ESPercentPerSecondHasCost", { format = "{2:output:ESPercentPerSecondCost}", { breakdown = "ESPercentPerSecondCost" }, { modName = { "ESCost", "Cost", "ESCostNoMult" }, cfg = "skill" }, }, },
 	{ label = "Rage Cost", color = colorCodes.RAGE, haveOutput = "RageHasCost", { format = "{0:output:RageCost}", { breakdown = "RageCost" }, { modName = { "RageCost", "Cost", "RageNoMult" }, cfg = "skill" }, }, },
 	{ label = "Rage per second", color = colorCodes.RAGE, haveOutput = "RagePerSecondHasCost", { format = "{2:output:RagePerSecondCost}", { breakdown = "RagePerSecondCost" }, { modName = { "RageCost", "Cost", "RageNoMult" }, cfg = "skill" }, }, },
+	{ label = "Armourbreak per hit", haveOutput = "ArmourBreakPerHit", { format = "{0:output:ArmourBreakPerHit}", { modName = "ArmourBreakPerHit", modType = "BASE"} }, },
```
You mean the displayed text ?

Nightblade replied:
Yeah "Armourbreak per hit", although now I'm thinking maybe it should be:  "Armour Break per hit" since it appears like that in-game?

MrHB212 replied:
The displayed text is too long, i will go with "Armour Break / hit"
![image](https://github.com/user-attachments/assets/774ef132-f18b-449a-8064-145174f5e6a7)
![image](https://github.com/user-attachments/assets/14a0183d-d434-4586-93d1-887d5046879d)

MrHB212 commented:
### Summary
- added section in calc tab to show armour break per hit
- added support for `Break X% increased Armour`
- added support for `Break X% increased Armour on Enemies affected by Ailments`
- added support for `apply_X_armour_break_on_hit`

In the calc tab it looks like this:
![image](https://github.com/user-attachments/assets/fcfc002e-6e71-43f2-9b73-4337969b56d4)

As far as i can tell it works as it should.

majochem commented on src/Modules/ModParser.lua:
```diff
@@ -3639,6 +3639,8 @@ local specialModList = {
 	["elemental damage with hits is lucky while you are shocked"] = { flag("ElementalLuckHits", { type = "Condition", var = "Shocked" }) },
 	["break (%d+)%% of armour on heavy stunning an enemy"] = { flag("Condition:CanArmourBreak", { type = "GlobalEffect", effectType = "Buff", effectName = "ArmourBreak" }) },
+	["break (%d+)%% increased armour"] = function(num) return { mod("ArmourBreakPerHit", "INC", num)} end,
+	["break (%d+)%% increased armour on enemies affected by ailments"] = function(num) return { mod("ArmourBreakPerHit", "INC", num, { type = "Condition", var = "AppliedAilmentsRecently"})} end,
```
I don't think this should be "AppliedAilmentsRecently" ?
The wording is "enemies affected by ailments", whether or not it was "applied recently" is not the deciding factor. It is whether there is any ailment on the enemy right now.

It should be possible to do this without a seperate config option as there are already existing conditions for ailments.

MrHB212 replied:
Youre right. Looking back i also wonder why i did it like that. I just changed it.

majochem commented on src/Modules/CalcOffence.lua:
```diff
@@ -345,6 +345,11 @@ function calcs.offence(env, actor, activeSkill)
 
+	-- Calculate armour break
+	local base = skillModList:Sum("BASE", nil, "ArmourBreakPerHit")
+	local inc = skillModList:Sum("INC", nil, "ArmourBreakPerHit")
```
I think you can use `calcLib.val()` for this to do it all in one step? Either way, this current iteration would not take "MORE" multipliers into account

MrHB212 replied:
I was unaware of the function. Thanks.
